### DISCUSSION TITLE
2.0 版本可能发生的变更

### DISCUSSION THREAD

**Discussion by LeeJim** [Tencent/tdesign-miniprogram#1761] [Show and tell]
你对 2.0 有什么期待？或者有什么建议？

**Comment by LeeJim** [comment]
# image-viewer 

- 移除 background-color 使用 CSS Variables 替代

**Reply by LeeJim** [reply]
简化 close 事件返回的信息

**Comment by LeeJim** [comment]
# Picker

- 避免暴露 picker-item 组件，直接使用 picker 的 columns 属性即可

**Comment by LeeJim** [comment]
# tabs
- animation.duration 单位从 s 改成 ms

**Reply by LeeJim** [reply]
类名中移除 placement 相关信息

**Comment by anlyyao** [comment]
## Link

- 移除 navigator相关改用 view 实现，navigatorProps 属性类型将出现不兼容变更

**Comment by LeeJim** [comment]
image-viewer，简化 close 事件返回的信息

**Comment by LeeJim** [comment]
# Loading

- size 默认值改成 20px

**Comment by anlyyao** [comment]
## Button

- 事件 tap 废弃，保留 click。（与基础事件同名，会导致事件被触发2次）

**Reply by jarmywang** [reply]
ghost 更名为：plain，与官方一致

**Comment by LeeJim** [comment]
# tabs

#2123 可以替换掉 space-evently 属性

**Comment by anlyyao** [comment]
2.0 流水线需要引入 spell check， 目前有多处拼写错误。（包括组件属性名称，涉及不兼容变更。）

**Comment by LeeJim** [comment]
# Toast
属性名变更：destory -> destroy

**Reply by LeeJim** [reply]
关联PR：https://github.com/Tencent/tdesign-miniprogram/pull/2185

**Reply by anlyyao** [reply]
#4624 
#1919

**Comment by LeeJim** [comment]
# Transition
属性名变更：destoryOnClose -> destroyOnClose

**Reply by LeeJim** [reply]
关联PR：https://github.com/Tencent/tdesign-miniprogram/pull/2185

**Comment by LeeJim** [comment]
# Popup

- 移除 defaultVisible
- 事件 visible-change 改成 close

**Comment by LeeJim** [comment]
# Upload

- 移除内部依赖的 grid，使用 grid css 来实现

**Comment by anlyyao** [comment]
# TabBar

- 使用 icon 插槽，不需要显示设置 icon === 'slot'（#2678）

**Comment by wflashy** [comment]
能加入像mobile-vue的table组件？
https://tdesign.tencent.com/mobile-vue/components/table

**Comment by jarmywang** [comment]
考虑移除表单类组件的受控组件与非受控组件概念，理由：此概念增加了用户的理解成本，小程序本身对此概念是淡化的，可以受控写法，绑定value，每次改变都setData，也可以仅仅通过onChange观察改变的值

**Comment by jarmywang** [comment]
# Cell、Radio、Checkbox

- 底部边框在容器内只有一个元素时不显示
- 统一borderless、bordered的属性命名

**Comment by cb360** [comment]
组件都适配一下skyline引擎吧

**Reply by cb360** [reply]
已经再用了，还有一部分没适配，没法用

**Comment by anlyyao** [comment]
建议 2.x 对于支持多类型+插槽的属性，其默认值做空值处理，或者对插槽做单独的说明，告知属性值优先级高于插槽，若已设置属性值，则插槽不生效

#3019

**Comment by Boomkaa** [comment]
### Input
---
补一个extra插槽[#1639]

**Comment by betavs** [comment]
# pull-down-refresh

- 移除 `usingCustomNavbar` 属性 (#3592)
